Legion/src/Assets/uimg.cpp

```diff
@@ -115,8 +115,18 @@ void RpakLib::ExtractUIImageAtlas(const RpakLoadAsset& Asset, const string& Path
 		string ImageName = string::Format("0x%x%s", img.Hash, (const char*)ImageExtension);
 
 		if (img.Path.Length() > 0)
-			ImageName = string::Format("%s%s", IO::Path::GetFileNameWithoutExtension(img.Path).ToCString(), (const char*)ImageExtension);
+			ImageName = string::Format("%s%s", img.Path.ToCString(), (const char*)ImageExtension);
 
-		tmp->Save(IO::Path::Combine(Path, ImageName), ImageSaveType);
+
+		const size_t last_slash_idx = ImageName.LastIndexOf('/');
+		string ImageDirectory = "";
+		if (std::string::npos != last_slash_idx)
+		{
+			ImageDirectory = ImageName.Substring(0, last_slash_idx);
+		}
```

Comment:
this is unnecessary - use [IO::Path::GetDirectoryName](https://github.com/r-ex/LegionPlus/blob/814acd4a791878efe0701a264b37eff867a3d26d/cppnet/cppkore/Path.h#L28)